New packages x11-libs/kcolorpicker and x11-libs/kimageannotator

dflogeras opened the pull request:
Rationale: https://bugs.gentoo.org/723758

These two packages will soon be dependencies of kde-apps/spectacle to add annotation capabilities.  I have been maintaining both in the GURU overlay since about May 2020.

a17r reviewed (changes requested):
Both packages have cmake build flags that are not handled by your ebuilds:
https://github.com/ksnip/kColorPicker/blob/master/CMakeLists.txt#L13
https://github.com/ksnip/kImageAnnotator/blob/master/CMakeLists.txt#L11
Please check whether they are useful (tests are working, examples actually install anything) and handle with `IUSE` as needed.

dflogeras commented:
Thanks for the thorough review, I'll address these comments in the next few days.

dflogeras commented:
What is the correct way to do this?  My original PR was two separate commits (one for kcolorpicker, one for kimageannotator).  In the past I have overwrote a single commit with a squash, then force pushed and the PR gets updated.  However, I am not sure what to do with multiple commits.  Sorry for my inexperience, but I want to do this in the best manner possible.

juippis commented:
You should read some git tutorial, but it goes something like this: 
Make new commits to fix any reviewed issues, so in the end you'll have 4 commits instead of 2. Then do 
`git rebase -i HEAD~4`
which opens an editor with your 4 commits. You can re-arrange those commits, do read the information provided in that window. So it looks like this:
```
pick 2854d8b23e4 x11-libs/kcolorpicker: New package
pick 2cea359719e x11-libs/kimageannotator: New package
pick 8d2f2b270ce my fixes to kcolorpicker
pick 195e1a3fde1 my fixes to kimageannotator
```
And you rearrange the commits in the editor, 
```
pick 2854d8b23e4 x11-libs/kcolorpicker: New package
pick 8d2f2b270ce my fixes to kcolorpicker
pick 2cea359719e x11-libs/kimageannotator: New package
pick 195e1a3fde1 my fixes to kimageannotator
```
and use `fixup` to fix your original commits:
```
pick 2854d8b23e4 x11-libs/kcolorpicker: New package
fixup 8d2f2b270ce my fixes to kcolorpicker
pick 2cea359719e x11-libs/kimageannotator: New package
fixup 195e1a3fde1 my fixes to kimageannotator
```
Save and exit, and that should do it. Please do read the difference with `squash` and `fixup` and use whichever you might require.

dflogeras commented:
@juippis Thanks very much, the missing piece for me was I didn't know that you could re-order like that.

I've addressed the comments, ready for a second look.  I found the packages did not install anything with BUILD_EXAMPLE so omitted it in IUSE.  I did add IUSE=test.  Please inspect the virtualx stuff; I _think_ I have it the way it should be but it could stand a second set of eyes.

Thanks.

a17r commented on x11-libs/kcolorpicker/metadata.xml:
```diff
@@ -0,0 +1,12 @@
+	<maintainer type="person">
+		<email>dflogeras2@gmail.com</email>
+		<name>David Flogeras</name>
+	</maintainer>
```
Missing
```
	<maintainer type="project">
		<email>proxy-maint@gentoo.org</email>
		<name>Proxy Maintainers</name>
	</maintainer>
```

a17r commented on x11-libs/kimageannotator/kimageannotator-0.3.2.ebuild:
```diff
@@ -0,0 +1,47 @@
+S="${WORKDIR}/${MY_P}"
+
+src_configure() {
+	mycmakeargs+=(
```
`local mycmakeargs`

a17r commented on x11-libs/kimageannotator/metadata.xml:
```diff
@@ -0,0 +1,10 @@
+	<maintainer type="person">
+		<email>dflogeras2@gmail.com</email>
+		<name>David Flogeras</name>
+	</maintainer>
```
Missing
```
	<maintainer type="project">
		<email>proxy-maint@gentoo.org</email>
		<name>Proxy Maintainers</name>
	</maintainer>
```

a17r commented on x11-libs/kimageannotator/metadata.xml:
```diff
@@ -0,0 +1,10 @@
+		<email>dflogeras2@gmail.com</email>
+		<name>David Flogeras</name>
+	</maintainer>
+	<longdescription lang="en">Tool for annotating images.
+	</longdescription>
```
Same as `DESCRIPTION`, so does not make sense to duplicate here. Also flagged by CI.

a17r commented on x11-libs/kimageannotator/kimageannotator-0.3.2.ebuild:
```diff
@@ -0,0 +1,47 @@
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+
+DEPEND="${RDEPEND}
+	test? ( dev-qt/qttest:5 )
+"
```
You're inheriting `RDEPEND`, so logically this block should really be below the `RDEPEND` block

a17r commented on x11-libs/kcolorpicker/kcolorpicker-0.1.4.ebuild:
```diff
@@ -0,0 +1,41 @@
+S="${WORKDIR}/${MY_P}"
+
+src_configure() {
+	mycmakeargs+=(
```
`local mycmakeargs`

a17r commented on x11-libs/kcolorpicker/kcolorpicker-0.1.4.ebuild:
```diff
@@ -0,0 +1,41 @@
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+
+DEPEND="${RDEPEND}
+	test? ( dev-qt/qttest:5 )
+"
```
You're inheriting `RDEPEND`, so logically this block should really be below the `RDEPEND` block

a17r commented on x11-libs/kcolorpicker/kcolorpicker-0.1.4.ebuild:
```diff
@@ -0,0 +1,41 @@
+}
+
+src_test() {
+	virtx default
```
`virtx cmake_src_test`

a17r commented on x11-libs/kimageannotator/kimageannotator-0.3.2.ebuild:
```diff
@@ -0,0 +1,47 @@
+}
+
+src_test() {
+	virtx default
```
`virtx cmake_src_test`

a17r reviewed (changes requested):
Thanks for your effort, let's get this finally merged.

One more thing: I'm not yet sold on the category. Even though they do currently depend on `X11` libraries, that may (hopefully) not be permanent, and for what they are doing they should be in `media-libs` where I recently added a similar package in commit 52b4976cdf0bbacdd54e460629ad6d57bb53dfb9.

dflogeras commented:
I have no strong opinion on the category; I think I just asked for suggestions in GURU and that's what we landed on. Is this as simple as git mv, change my dep in the ebuild, then force push?  I guess renaming this PR is also in order.

Finally, I see that kimageannotator was bumped to 0.4.x (with a 0.4.1 pending).  I feel we should move ahead with this (0.3.2), then bump it once in tree, unless you require a newer version to start or it creates more work than it's worth.  Thoughts?

a17r commented:
Renaming now is much easier than after merging this PR, so yes just go for 'git mv' and fix the dep, force-push.

The changes for 0.4.0 look worth it for you to move to that version right away, I don't see new deps, but link interface fixes. Will not delay this PR. ;)

No need to change the PR title, just `git --amend` your git summaries (or do it using `git rebase -i` interactively.

dflogeras commented:
Sounds good.  Except, I tried to bump to 0.4.0 and got a bunch of tests failing to compiled.  After hunting it down, I've submitted a PR.  I guess our options (depending on how much of a hurry you are in) are:

a) Stick with 0.3.2, quick and working
b) Wait to see if upstream accepts my fix, or some derivative, and releases 0.4.1 (slower)
c) Apply my patch in-tree and figure out b) at a later date.  Compromise.

What's your time-frame for when this starts to get in the way of plasma?

dflogeras commented:
Also, the PR is https://github.com/ksnip/kImageAnnotator/pull/195 for posterity

a17r commented:
Your patch looks good, just add it to the ebuild with a reference to the upstream PR#

a17r commented:
git messages aren't fixed yet
